lms/djangoapps/courseware/access.py

```diff
@@ -257,6 +257,21 @@ def _can_enroll_courselike(user, courselike):
         log.warning("ENROLLMENT DENIED in %s because @ in username %s" % (course_key, user.username))
         return ACCESS_DENIED
 
+    if settings.FEATURES.get('RESTRICT_ENROLL_SOCIAL_PROVIDERS'):
+        try:
+            social_providers = user.social_auth.all()
+        except AttributeError:
+            social_providers = []
+        allowed_providers = settings.FEATURES.get('RESTRICT_ENROLL_SOCIAL_PROVIDERS')
+        if social_providers and not any(
+            [any([social_provider.uid.startswith(allowed_provider)
```

Comment:
I might be confusing something but I think this should be `social_provider.provider.startswith(allowed_provider)`

Reply:
No, because the `provider` attribute is always `tpa-saml` whereas the UID is of the form `mit-kerberos:tmacey`

Reply:
sounds good 👍